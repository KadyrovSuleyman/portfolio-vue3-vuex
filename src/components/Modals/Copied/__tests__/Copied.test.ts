/* eslint-disable no-param-reassign */
import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import Notification from '../Copied.vue';

jest.mock('../state.ts');

let store: Store<any>;
beforeEach(() => {
  store = createStore<any>({
    state: {
      isShown: false,
    },
    mutations: {
      change: (state, obj: { [name: string]: boolean | string }) => {
        Object.keys(obj).forEach((index) => {
          state[index] = obj[index];
        });
      },
    },
  });
});

let element: HTMLElement;
beforeEach(() => {
  element = document.createElement('div');
  element.className = 'app';
  document.body.appendChild(element);
});
afterEach(() => {
  document.body.outerHTML = '';
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('copied renders', async () => {
  wrapper = mount(Notification, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  // expect(wrapper.findComponent('.copied-notification').exists()).toBeFalsy();
  expect(element.outerHTML).toMatchSnapshot();

  // store.commit('change', {
  //   isShown: true,
  // });
  store.state.isShown = true;
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();
  expect(element.outerHTML).toMatchSnapshot();

  // expect(wrapper.findComponent('.copied-notification').exists()).toBeTruthy();

  // expect(wrapper.findComponent('.copied-span').exists()).toBeTruthy();
  // expect(wrapper.findComponent('.copied-span').text()).toBe('Copied!');
});

// it('watchs props changes', async () => {
//   store.state.isShown = true;
//   wrapper = mount(Notification, {
//     global: { plugins: [store] },
//     props: {
//       target: '.app',
//     },
//   });
//   expect(wrapper.findComponent('.copied-notification').classes()).toEqual(['copied-notification']);

//   await wrapper.setProps({
//     ...wrapper.props,
//     mods: {
//       selected: true,
//     },
//   });
//   expect(wrapper.findComponent('.copied-notification').classes()).toEqual([
//     'copied-notification',
//     'copied-notification__selected',
//   ]);

//   await wrapper.setProps({
//     ...wrapper.props,
//     mods: {
//       selected: false,
//     },
//   });
//   expect(wrapper.findComponent('.copied-notification').classes()).toEqual(['copied-notification']);
// });

// it('closes', async () => {
//   wrapper = mount(Notification, {
//     global: { plugins: [store] },
//     props: {
//       target: '.app',
//     },
//   });

//   expect(wrapper.findComponent('.copied-notification').exists()).toBeFalsy();

//   store.commit('change', {
//     isShown: true,
//   });
//   await wrapper.vm.$nextTick();
//   expect(store.state.isShown).toBeTruthy();
//   expect(wrapper.findComponent('.copied-notification').exists()).toBeTruthy();

//   wrapper.findComponent('.notification-button').trigger('click');
//   await wrapper.vm.$nextTick();
//   expect(store.state.isShown).toBeFalsy();
//   expect(wrapper.findComponent('.copied-notification').exists()).toBeFalsy();
// });
