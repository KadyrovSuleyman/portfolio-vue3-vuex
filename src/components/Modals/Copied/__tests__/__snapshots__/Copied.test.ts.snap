// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copied renders 1`] = `
<div class="app">
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;

exports[`copied renders 2`] = `
<div class="app">
  <transition-stub>
    <div class="copied-notification"><span class="copied-span"> Copied! </span><button class="notification-button notification-button__icon_cross_white"></button></div>
  </transition-stub>
</div>
`;
